src/Microsoft.Health.Fhir.Core/Features/Search/SearchValues/ReferenceSearchValue.cs

```diff
@@ -85,8 +86,8 @@ public bool Equals([AllowNull] ISearchValue other)
 
             return Kind == referenceSearchValueOther.Kind &&
                    BaseUri == referenceSearchValueOther.BaseUri &&
-                   ResourceType.Equals(referenceSearchValueOther.ResourceType, StringComparison.OrdinalIgnoreCase) &&
-                   ResourceId.Equals(referenceSearchValueOther.ResourceId, StringComparison.OrdinalIgnoreCase);
+                   string.Equals(ResourceType, referenceSearchValueOther.ResourceType, StringComparison.OrdinalIgnoreCase) &&
```

Comment:
Is it possible for these values to be null? Should we then check for the null case as you have done in other types?

Reply:
If I understand your ask correctly, string.Equals will handle null values for e..g if ResourceType is null, then null will be compared with  referenceSearchValueOther.ResourceType for equality and return true/false accordingly